torch/distributions/beta.py

```diff
@@ -28,7 +29,7 @@ class Beta(ExponentialFamily):
     has_rsample = True
 
     def __init__(self, concentration1, concentration0, validate_args=None):
-        if isinstance(concentration1, Number) and isinstance(concentration0, Number):
+        if isinstance(concentration1, Real) and isinstance(concentration0, Real):
```

Comment:
Why change from Number to Real is needed here? I.e. can't number be always cast to float?

Reply:
Read the doc, number can be a complex value